File: C++/Ders/Ders-3/Student.cpp
#include "Student.h"

Student::Student() {
	this->department = "abc";
}

Student::Student(std::string name, std::string surname, int age, std::string department) 
	: department(department), Person(name, surname, age){}

std::string Student::getDepartment() {
	return this->department;
}

void Student::setDepartment(std::string department) {
	this->department = department;
}

std::ostream& operator<< (std::ostream& output, Student& obj) {
	output << "Name: " << obj.getName() << "\nSurname: " << obj.getSurname() << "\nAge: " << obj.getAge() << "\nDepartment: " << obj.getDepartment() << "\n";
	return output;
}

File: JAVA/Ödev/Odev-1/src/Student.java
public class Student extends Person{
    private String department;

    public Student(String name, String surname, int ID, String department) {
        super(name, surname, ID);
        this.department = department;
    }

    @Override
    void printInfo() {
        System.out.println("Name: " + getName() + "\nSurname: " + getSurname() + "\nID: " + getID() + "\nDepartment: " + getDepartment());
    }

    public String getDepartment() {
        return department;
    }

    public void setDepartment(String department) {
        this.department = department;
    }
}



